src/device.cpp

```diff
@@ -170,14 +170,22 @@ device::device(std::shared_ptr<context> ctx,
                const platform::backend_device_group group,
                bool device_changed_notifications)
     : _context(ctx), _group(group), _is_valid(true),
-      _device_changed_notifications(device_changed_notifications)
+      _device_changed_notifications(device_changed_notifications),
+      _is_alive( true )
 {
     _profiles_tags = lazy<std::vector<tagged_profile>>([this]() { return get_profiles_tags(); });
 
     if (_device_changed_notifications)
     {
-        auto cb = new devices_changed_callback_internal([this](rs2_device_list* removed, rs2_device_list* added)
+        std::weak_ptr< bool > weak = std::make_shared< bool >( _is_alive );
```

Comment:
This shouldn't work!
Make the shared - good.
Make the weak - good.
Destroy the shared - because nobody's holding it!
Now in the callback, `weak.lock()` has no strong pointer alive, it will return null...
@OhadMeir please review

Reply:
You are right.
I think we missed that 
```
bool _is_alive;
```
should be
```
std::shared_ptr<bool> _is_alive;
```

and add the `_alive(make_shared<bool>(true))..` in the Ctor or something like that.